napari/layers/_layer_actions.py

```diff
@@ -64,7 +64,7 @@ def _convert(ll: LayerList, type_: str) -> None:
         else:
             data = lay.data
         new_layer = Layer.create(data, lay._get_base_state(), type_)
```

Comment:
The problem is here. Both layers share the data. It means That editing labels will edit the image. Another option is to make copy, but this is expensive in the context of people who load labels as image layer and need to convert it labels. 

I think that we should preserve only the Shapes layer as operation is reversible for `Image` → `Labels` 

Another option could be implementing "copy on write mechanism" but for this, we need to count how many layers point to a given array.

Reply:
We have one action for all layer types. I'm not sure if it is optimal...